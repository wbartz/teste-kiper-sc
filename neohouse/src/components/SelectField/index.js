import PropTypes from 'prop-types';
import React, { useCallback } from 'react';
import './index.scss';

const SelectField = ({
  onChange,
  value,
  name,
  label,
  options,
  register,
  errormsg,
  disabled,
}) => {
  const handleChange = useCallback(text => onChange(text), [onChange]);

  return (
    <div
      className={`input-field select input-outlined ${errormsg && 'invalid'}`}
    >
      <select
        disabled={disabled}
        className="browser-default"
        ref={register}
        value={value}
        name={name}
        id={name}
        onChange={event => {
          handleChange(event.target.value);
        }}
        data-testid={name}
      >
        {options.map(line => (
          <option key={`${line.label}-${line.value}`} value={line.value}>
            {line.label}
          </option>
        ))}
      </select>
      {label && <label htmlFor={name}>{label}</label>}
      {errormsg && (
        <span className="error">
          <i className="material-icons left">error</i>
          {errormsg}
        </span>
      )}
    </div>
  );
};

SelectField.propTypes = {
  onChange: PropTypes.func,
  name: PropTypes.string,
  value: PropTypes.string,
  label: PropTypes.string,
  options: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  errorMsg: PropTypes.string,
  disabled: PropTypes.bool,
  register: PropTypes.oneOfType([PropTypes.object]),
};

SelectField.defaultProps = {
  onChange: () => {},
  value: '',
  name: '',
  label: '',
  errorMsg: '',
  disabled: false,
  register: null,
};

export default SelectField;
