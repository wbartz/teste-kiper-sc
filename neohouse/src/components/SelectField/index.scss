@import '../../styles/variables';

.input-field.select {
  select {
    border-color: $color-lighter-gray;
    color: $color-black;
    font-size: 16px;
  }

  label {
    font-family: $font-family-alternative;
    font-size: 16px;
    left: 0px;
    top: -36px;
    position: absolute;
    background-color: #ffffff;
    padding: 0 5px;
    text-transform: lowercase;
  }

  label::first-letter {
    text-transform: uppercase;
  }

  .postfix {
    position: absolute;
    font-size: 2rem;
    transition: $transition-color;
    top: $font-size-regular;
    right: -30px;
  }

  span.error {
    font-family: $font-family-alternative;
    font-size: 12px;
    color: $color-error;

    i {
      font-size: 16px;
    }
  }
}

.input-field.invalid {
  input {
    border-color: $color-error !important;
  }
}
