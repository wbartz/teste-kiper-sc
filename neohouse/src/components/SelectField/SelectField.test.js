import { shallow } from 'enzyme';
import React from 'react';
import SelectField from '.';

describe('SelectField', () => {
  const props = {
    value: '',
    name: 'SelectField',
    onChange: jest.fn(),
    onKeyPress: jest.fn(),
    label: 'SelectField',
    options: [{ value: 'teste', label: 'teste' }],
  };
  const wrapper = shallow(<SelectField {...props} />);

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
