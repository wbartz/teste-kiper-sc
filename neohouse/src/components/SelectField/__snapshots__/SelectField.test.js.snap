// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField render without crashing 1`] = `ShallowWrapper {}`;
